kratos/linear_solvers/preconditioner.h

```diff
@@ -403,6 +402,63 @@ inline std::ostream& operator << (std::ostream& OStream,
 ///@}
 
 
+template< typename TSparseSpace, typename TlocalSpace>
```

Comment:
same argument as for the linear_solver_base. I am putting it in the same archive to have it available where the preconditioenr is available